src/components/input.js

```diff
@@ -146,44 +146,48 @@ export const SearchInput = ({ value, onChange, ...props }) => {
 
 export const DelayedSearchInput = withDebouncedChange(SearchInput)
 
-export const NumberInput = ({ onChange, onBlur, min = -Infinity, max = Infinity, onlyInteger = false, isClearable = true, tooltip, value, ...props }) => {
+export const NumberInput = forwardRefWithName('NumberInput', ({ onChange, onBlur, min = -Infinity, max = Infinity, onlyInteger = false, isClearable = true, tooltip, value, ...props }, ref) => {
   // If the user provided a tooltip but no other label, use the tooltip as the label for the input
   useLabelAssert('NumberInput', { tooltip, ...props, allowId: true, allowTooltip: true })
 
   const [internalValue, setInternalValue] = useState()
 
-  const numberInputChild = div([input(_.merge({
-    type: 'number',
-    'aria-label': Utils.getAriaLabelOrTooltip({ tooltip, ...props }),
-    className: 'focus-style',
-    min, max,
-    value: internalValue !== undefined ? internalValue : _.toString(value), // eslint-disable-line lodash-fp/preferred-alias
-    onChange: ({ target: { value: newValue } }) => {
-      setInternalValue(newValue)
-      // note: floor and clamp implicitly convert the value to a number
-      onChange(newValue === '' && isClearable ? null : _.clamp(min, max, onlyInteger ? _.floor(newValue) : newValue))
-    },
-    onBlur: (...args) => {
-      onBlur && onBlur(...args)
-      setInternalValue(undefined)
-    },
-    style: {
-      ...styles.input,
-      width: '100%',
-      paddingLeft: '1rem',
-      paddingRight: '0.25rem',
-      fontWeight: 400,
-      fontSize: 14,
-      backgroundColor: props.disabled ? colors.dark(0.25) : undefined
-    }
-  }, props))])
+  const numberInputChild = div([input({
+    ..._.merge({
```

Comment:
Do we need `merge` at all in this case? Could we just use the spread operator?

Reply:
It looks like `merge` is used for overriding the component's default styles with the style prop, but that could also be done with the spread operator.